### ISSUE TITLE
Android build missing jansson_private_config.h

### ISSUE THREAD

**Issue by JGregoryAtiba** [akheron/jansson#406]
I've added Jansson as a repo manifest project into `external/jansson` as part of my AOSP build.  Because the project is checked out from Github, there's no `jansson_private_config.h`, so my build fails.  The README says to run `autoreconf` to generate `configure`, which presumably can then be run to generate `jansson_private_config.h`.  But, that's not usually how things are done with AOSP.

I have two questions:

1. How platform-dependent are the contents of `jansson_private_config.h`?
2. Would it be possible to add an AOSP-specific version of this header to the `android/` directory, alongside the `jansson_config.h` currently there?

**Comment by mxi1** [comment]
The `jansson_private_config.h` file could be borrowwed from the linux x64 platform build process. You can follow the normal process ( `autoreconf -i` , then `.configure` ), until you find the newly-created `jansson_private_config.h` , you can put this header file to your `android` directory. 

@coreyfarrell,  using this private config header file, I can build jansson for the android platform. However, there're still several places where could be improved for the current jansson codebase.

1.  The `android.mk` file is out of date, and Google has been prompting new `android.bp` file to replace the `.mk` files;  Google provided one command-line utility to convert;
2.  I prefer to seprate the `jansson.h` file from the existing `src` directory to one newly-created `include` directory; since all other software only dpend on this header file and the `libjansson.so` library.

If you don't mind, I can prepare the code changes and raise one PR to this repo. Thank you.

**Comment by akheron** [comment]
It would probabjy make sense to remove `Android.mk`. It hasn't been updated in 7 years :grimacing:

**Comment by mxi1** [comment]
> It would probably make sense to remove `Android.mk`. It hasn't been updated in 7 years grimacing

I agree with you. The `Android.bp` file is better from Google's viewpoint.  I have converted the `Android.mk` file to new `Android.bp` file.  Besides, I created the `android_private_config.h` file based on the file generated on my ubuntu x64 pc. 

Another thing, I moved the `janson.h` file out of the `src` directory, to the newly-created directory `include`. The applications depending `jansson` doesn't need to include all header files under the `src` directory. the `include` directory should be enough to them. 

All the changed mentioned above is in [the android branch of my repo](https://github.com/mxi1/jansson/tree/android). 
I didn't create the PR, because I suddenly think these changes should be put two PRs: one for `android` support; the other for `janson.h` location change.
